packages/model-viewer/src/features/scene-graph/texture.ts

```diff
@@ -45,12 +50,17 @@ export class Texture extends ThreeDOMElement implements TextureInterface {
         {};
     this[$sampler] = new Sampler(onUpdate, sampler, correlatedTextures);
 
-    if (gltf.images != null && imageIndex != null) {
+    if (gltf.images != null && imageIndex != null && imageIndex !== -1) {
       const image = gltf.images[imageIndex];
 
       if (image != null) {
         this[$source] = new Image(onUpdate, image, correlatedTextures);
       }
+    } else if (imageIndex === -1) {
```

Comment:
If you move this block ahead, you can just put an early return and skip and extra check.

Reply:
Done.